### PR TITLE
Add some process_inbox calls for the `matching_engine` end to end test

### PR DESCRIPTION
**Pull request by MathieuDutSik**
## Motivation

Replacement of the error `failed to post query` with a more precise error message revealed the origin of some errors.

## Proposal

Two entries of `process_queries` are added to the `matching_engine`.
It is possible that the iterative loop for `make_application` needs to be eliminated.

## Test Plan

An improved CI is the objective of this work.

## Release Plan

Not relevant.

## Links

- [reviewer checklist](https://github.com/linera-io/linera-protocol/blob/main/CONTRIBUTING.md#reviewer-checklist)

### CODE REVIEW

File: linera-service/tests/end_to_end_tests.rs [hunk 1]
```diff
@@ -1436,6 +1436,9 @@ async fn test_wasm_end_to_end_matching_engine(config: impl LineraNetConfig) -> R
     node_service_admin
         .request_application(&chain_admin, &token1)
         .await?;
+    node_service_admin.process_inbox(&chain_admin).await?;
+    node_service_a.process_inbox(&chain_a).await?;
+    node_service_b.process_inbox(&chain_b).await?;
```
Comment (ma2bd) [inline linera-service/tests/end_to_end_tests.rs:1441]: Yes these ones look necessary.

File: linera-service/tests/end_to_end_tests.rs [hunk 2]
```diff
@@ -1500,6 +1503,9 @@ async fn test_wasm_end_to_end_matching_engine(config: impl LineraNetConfig) -> R
             &[token0.forget_abi(), token1.forget_abi()],
         )
         .await?;
+    node_service_admin.process_inbox(&chain_admin).await?;
+    node_service_a.process_inbox(&chain_a).await?;
+    node_service_b.process_inbox(&chain_b).await?;
```
Comment (ma2bd) [inline linera-service/tests/end_to_end_tests.rs:1508]: These ones look too early. We want to make sure that the RequestApplication messages are processed, then their answers.

Comment (MathieuDutSik) [reply linera-service/tests/end_to_end_tests.rs:1508]: Ok, I changed the position of the `process_inbox` statements.

File: linera-service/tests/end_to_end_tests.rs [hunk 3]
```diff
@@ -1521,6 +1524,9 @@ async fn test_wasm_end_to_end_matching_engine(config: impl LineraNetConfig) -> R
             .make_application(&chain_b, &application_id_matching)
             .await?,
     );
+    node_service_admin.process_inbox(&chain_admin).await?;
+    node_service_a.process_inbox(&chain_a).await?;
+    node_service_b.process_inbox(&chain_b).await?;
```
[the remark this answers is not in the document]
Comment (ma2bd) [reply linera-service/tests/end_to_end_tests.rs:1529]: @MathieuDutSik The `process_inbox` are useful when they happen between `request_application` and `make_application`

[the remark this answers is not in the document]
Comment (MathieuDutSik) [reply linera-service/tests/end_to_end_tests.rs]: Ok, I changed the position of the `process_inbox` for `a` and `b` and eliminated it for `admin`.

**Review by afck — APPROVED** [review]

**Comment by MathieuDutSik** [comment]
This attempt was in the right direction but was superseded by PR #2144

**Pull request closed without merging**